phoenix-core/src/main/java/org/apache/phoenix/schema/PTableImpl.java

```diff
@@ -779,11 +784,20 @@ public void setValue(PColumn column, byte[] byteValue) {
         @Override
         public void delete() {
             newMutations();
-            Delete delete = new Delete(key);
-            for (PColumnFamily colFamily : families) {
-            	delete.addFamily(colFamily.getName().getBytes(), ts);
+            if (PTableImpl.this.isTransactional()) {
+                Put delete = new Put(key);
```

Comment:
Comment here that we're using the Tephra column family delete marker here to prevent the translation and early update on the server (similar to the storeNulls comment earlier).